### PR TITLE
feat(server): add prefix cache to paged concurrency (qwen AR)

### PR DESCRIPTION
**Pull request by Graffioh**
## Summary

- Enable copied in-memory prefix checkpoints for Qwen continuous batching with paged attention. Full-prefill and disk snapshots remain disabled in paged mode.
- Keep token matching, boundary selection, pinning, LRU, and hit accounting in PrefixCache while Qwen owns checkpoint payloads, page-table restore, and recurrent state.
- Introduce an opaque checkpoint/ticket protocol and a move-only capture transaction so the scheduler never manipulates backend snapshot storage.
- Gather and scatter logical K/V rows through fresh sequence-owned pages, copy the matching recurrent slab, coalesce consecutive physical blocks, and publish replacements atomically.

## Correctness review

The pre-draft review fixed these lifecycle hazards:

- stale restore cleanup now happens before busy or failed admission returns;
- unsupported engines use cold admission without deleting a valid checkpoint;
- malformed or mismatched events can abort only their own capture ticket;
- fatal step failures discard the pending destination instead of preserving potentially mutated state;
- failed replacement leaves the incumbent checkpoint intact;
- cache hits and MRU promotion are recorded only after a validated restore;
- exact-prompt checkpoints cannot mask a shorter restorable prefix when next-token logits are unavailable;
- tensor type, contiguity, extents, vector topology, recurrent slab shape, and per-layer tensor pairs are validated before destination writes.

## Verification

- HIP Release build of dflash_server
- test_server_unit: 418 passed
- test_seq_slot_manager: 186 checks
- test_parallel_prefix_txn: 21 checks
- test_seq_engine_contract: 19 checks
- test_recurrent_snapshot: 2 passed
- git diff --check

## Draft gates

- Measure capture and restore pause time plus snapshot memory at realistic long-prefix sizes while decode traffic is live.
- Run real Qwen GPU parity for a partial final page and a forced mid-message boundary, restoring into non-identity fresh pages.
- Decide the copied-checkpoint byte budget before broad production enablement.

This PR targets main as the AR ownership/mechanism base. Open speculative concurrency PRs #642 and #654 overlap in SeqEngine, Qwen, scheduler, and target-graph files; they should consume this opaque prefix-store boundary during their rebase rather than duplicate snapshot ownership.

<a href="https://cubic.dev/pr/Luce-Org/lucebox/pull/655?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by Graffioh** [comment]
@cubic-dev-ai review this PR
